### PR TITLE
Introduce different color panels on UI

### LINKED ISSUE #191: Introduce different color panels on UI
**Description**

As discussed on the SIG Content, we noticed that different color panels would be helpful to grab the users' attention when they read the Kyma documentation. For now, we only have NOTE to highlight some issues, but **Caution** and **Tip** would also be useful.

Apart from the technical implementation, the documentation will also have to be updated. It would be helpful to create a document in the `community` repo that explains how to add such panels in the documentation and when to use them.

**Reasons**

This implementation will not only improve the user experience but it will also allow us to focus users' attention on matters that we'd like to highlight. 

**Attachments**

These are the exemplary panels used on the [Yaas website](https://devportal.yaas.io/tools/documentationsdk/index.html#Panels). As discussed on the SIG Content, apart from Notes that we already have, we'd be interested in **Caution** and **Tip**. 

panels

Find the descriptions of different UI Panels in the [official SAP documentation](https://help.sap.com/viewer/DRAFT/e33c591ae4494a659a3f5f983c9d1161/PROD/en-US/bc7ef30971c04857b5d7cb50193065c1.html).

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- add _**TIP**_ and _**CAUTION**_ color panels for docs, like existing >**_NOTE_** blocks

**Related issue(s)**

Resolves #191

### CODE REVIEW

**Comment by aerfio** [comment]
image

**Comment by aerfio** [comment]
Version that is consistent with designs 
image

File: src/components/shared/Panel/styled.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import styled from "@styled";
+
+const tipColor = "#49C7A0";
```
[the remark this answers is not in the document]
Comment (aerfio) [reply src/components/shared/Panel/styled.ts]: Added!

**Comment by aerfio** [comment]
- [ ] add docs about this functionality

**Comment by aerfio** [comment]
For docs see https://github.com/kyma-project/community/pull/234

**Comment by aerfio** [comment]
Right now this fixes one case from https://github.com/kyma-project/website/issues/236 - what looks right now like this at kyma-project.io 
image

and like this in my PR: 
image

File: src/components/shared/NotePanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+import { PanelWrapper, PanelContent } from "./styled";
+
+export type PanelPropsType = "note" | "tip" | "caution" | undefined;
```
Comment (magicmatatjahu) [inline src/components/shared/NotePanel/index.tsx]: `export type NotePanelPropsType = "note" | "tip" | "caution" | undefined;`

File: src/components/shared/NotePanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export interface PanelProps {
```
Comment (magicmatatjahu) [inline src/components/shared/NotePanel/index.tsx]: `export interface NotePanelProps`

File: src/components/shared/NotePanel/styled.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    case "caution":
+      return css`
+        ${data => data.theme.colors.border.quaternary}
+      `;
+    default:
+      return "unset";
+  }
+};
+
+export const PanelWrapper = styled.blockquote`
```
Comment (magicmatatjahu) [inline src/components/shared/NotePanel/styled.ts]: `export const NotePanelWrapper = styled.blockquote`

File: src/components/shared/NotePanel/styled.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  margin-left: 0;
+  margin-right: 0;
+  padding: 16px;
+  & > div > p > strong:first-child {
+    color: ${(props: PanelProps) => getColorFromType(props)};
+  }
+  border-left: 3px solid ${(props: PanelProps) => getColorFromType(props)};
+`;
+
+export const PanelContent = styled.div`
```
Comment (magicmatatjahu) [inline src/components/shared/NotePanel/styled.ts]: `export const NotePanelContent = styled.div`

File: src/components/shared/ReactMarkdown/components/Blockquote.tsx [hunk 6]
```diff
@@ -1,12 +1,29 @@
...
+      child.props.children[0].props.children[0].props.value;
 
-export const BlockQuote: React.FunctionComponent = ({ children }) => (
-  <StyledBlockQuote>{children}</StyledBlockQuote>
-);
+    return type && type.replace(":", "").toLowerCase();
+  };
+
+  const createPanels = (elem: any) =>
+    elem.map((child: any, index: number) => {
```
Comment (magicmatatjahu) [inline src/components/shared/ReactMarkdown/components/Blockquote.tsx]:
``` js
elem.map((child: any, index: number) => (
        <NotePanel type={getPanelType(child)} key={index}>
          {child}
        </NotePanel>
)
```

**Pull request merged**